### ISSUE TITLE
[iOS 12] RXCoreData crash on iOS 12

### ISSUE THREAD

**Issue by hoangtuanfithou** [RxSwiftCommunity/RxCoreData#37]
it run nicely on iOS 11.
then check on iOS 12 -> crash. (mostly when fetch data)
more information will be updated
<img width="1005" alt="screen shot 2018-06-20 at 6 45 38 pm" src="https://user-images.githubusercontent.com/93890/41653898-2855e3c6-74ba-11e8-9ecc-4354defc8f98.png">

**Comment by bobgodwinx** [comment]
I think we should for a `GM` before updating anything. It's still a beta version.

**Comment by grigorye** [comment]
Just in case, I was able to mitigate it by enforcing Swift optimization (-Osize) for RxCoreData target. If you use CocoaPods, you can get it by appending the following to your Podfile:

```
post_install do |installer|
  # Enforce optimization for Xcode 10 beta.
  installer.pods_project.targets.each do |target|
    if target.name =~ /^RxCoreData$/
      puts("Enforcing Swift optimization for RxCoreData/Xcode 10 beta")
      target.build_configurations.each do |configuration|
        configuration.build_settings['SWIFT_OPTIMIZATION_LEVEL'] = '-Osize'
      end
    end
  end
end
```

**Comment by hoangtuanfithou** [comment]
For me build is fine on Xcode 10, just crashed when run.

**Comment by bobgodwinx** [comment]
@grigorye @hoangtuanfithou Let's see what happens later with the code size optimization option. It's pretty new and was introduced in [here in Swift 4.1](https://swift.org/blog/osize/). If necessary we could activate it in the next version.

**Comment by grigorye** [comment]
@bobgodwinx Just in case, "-Osize" was just quick hack - I just noticed that the crash doesn't happen in Release builds and hence I tried to enable optimization. Probably it doesn't crash with other optimization options. Anyway, I completely agree that this is just a workaround and we should wait for GM. More or less it looks like a bug in (beta of) Swift compiler that hopefully would be fixed in the upcoming betas (beta 2 still crashes).
